python/common/org/python/types/Str.java

```diff
@@ -708,10 +708,21 @@ private long toPositiveIndex(long index) {
     }
 
     @org.python.Method(
-            __doc__ = ""
+            __doc__ = "string.isalnum() -> bool\n" +
+                    "\n" +
+                    "Returns false when the string has atleast ONE character which is\n" +
+                    "not an alphanumeric or if the string is empty.\n"
     )
     public org.python.Object isalnum() {
-        throw new org.python.exceptions.NotImplementedError("isalnum() has not been implemented.");
+        if (!this.value.isEmpty()) {
```

Comment:
I also wanted to do it like that but as mentioned in above comment, I took `isalpha()` as a reference (before refactoring).
I'll change it to this logic flow.
Thank you for reviewing my PR :)